app/server/lib/ExportTableSchema.ts

```diff
@@ -0,0 +1,122 @@
+import * as express from 'express';
+import {ActiveDoc} from 'app/server/lib/ActiveDoc';
+import {DownloadOptions, exportTable} from 'app/server/lib/Export';
+import { ApiError } from 'app/common/ApiError';
+
+interface ExportColumn {
+  id: number;
+  colId: string;
+  label: string;
+  type: string;
+  widgetOptions: any;
+  description?: string;
+  parentPos: number;
+}
+
+export async function downloadTableSchema(
+  activeDoc: ActiveDoc,
+  req: express.Request,
+  options: DownloadOptions
+) {
+  const {tableId} = options;
+  if (!activeDoc.docData) {
+    throw new Error('No docData in active document');
+  }
+
+  // Look up the table to make a CSV from.
+  const settings = activeDoc.docData.docSettings();
+  const tables = activeDoc.docData.getMetaTable('_grist_Tables');
+  const tableRef = tables.findRow('tableId', tableId);
+
+  if (tableRef === 0) {
+    throw new ApiError(`Table ${tableId} not found.`, 404);
+  }
+
+  const data = await exportTable(activeDoc, tableRef, req);
+  const tableSchema = columnsToTableSchema(tableId, data, settings.locale);
+  return tableSchema;
+}
+
+function columnsToTableSchema(
+  tableId: string,
+  {tableName, columns}: {tableName: string, columns: ExportColumn[]},
+  locale: string,
+) {
+  return {
+    name: tableId.toLowerCase().replace(/_/g, '-'),
+    title: tableName,
+    schema: {
+      fields: columns.map(col => ({
+        name: col.label,
+        ...(col.description ? {description: col.description} : {}),
+        ...buildTypeField(col, locale),
+      })),
+    }
+  };
+}
+
+function buildTypeField(col: ExportColumn, locale: string) {
+  const type = col.type.split(':', 1)[0];
+  switch (type) {
+    case 'Text':
+      return {
+        type: 'string',
+        format: col.widgetOptions.widget === 'HyperLink' ? 'uri' : 'default',
```

Comment:
Widget options evolved in untyped javascript code.  I see a start at adding a type for it in ExcelFormatter. If it doesn't become a quagmire, it'd be very helpful to move that to app/common/WidgetOptions.ts and extend a bit further as you need it, so we aren't adding still more untyped code.